packages/flutter/test/widgets/list_wheel_scroll_view_test.dart

```diff
@@ -0,0 +1,390 @@
+            itemExtent: 20.0,
+            children: <Widget>[],
+          ),
+        )
+      );
+      fail('Expected failure with negative diameterRatio');
+    } on AssertionError {
+      // Exception expected.
+    }
+  });
+
+  testWidgets('ListWheelScrollView needs positive item extent', (WidgetTester tester) async {
+    try {
+      await tester.pumpWidget(
+        new Directionality(
+          textDirection: TextDirection.ltr,
+          child: new ListWheelScrollView(
+            itemExtent: null,
+            children: <Widget>[new Container()],
+          ),
+        )
+      );
+      fail('Expected failure with null itemExtent');
+    } on AssertionError {
+      // Exception expected.
+    }
+  });
+
+  testWidgets("ListWheelScrollView takes parent's size with small children", (WidgetTester tester) async {
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new ListWheelScrollView(
+          // Inner children smaller than the outer window.
+          itemExtent: 50.0,
+          children: <Widget>[
+            new Container(
+              height: 50.0,
+              color: const Color(0xFFFFFFFF),
+            ),
+          ],
+        ),
+      )
+    );
+    expect(tester.getTopLeft(find.byType(ListWheelScrollView)), const Offset(0.0, 0.0));
+    // Standard test screen size.
+    expect(tester.getBottomRight(find.byType(ListWheelScrollView)), const Offset(800.0, 600.0));
+  });
+
+  testWidgets("ListWheelScrollView takes parent's size with large children", (WidgetTester tester) async {
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new ListWheelScrollView(
+          // Inner children 5000.0px.
+          itemExtent: 50.0,
+          children: new List<Widget>.generate(100, (int index) {
+            return new Container(
+              height: 50.0,
+              color: const Color(0xFFFFFFFF),
+            );
+          }),
+        ),
+      )
+    );
+    expect(tester.getTopLeft(find.byType(ListWheelScrollView)), const Offset(0.0, 0.0));
+    // Still fills standard test screen size.
+    expect(tester.getBottomRight(find.byType(ListWheelScrollView)), const Offset(800.0, 600.0));
+  });
+
+  testWidgets("ListWheelScrollView children can't be bigger than itemExtent", (WidgetTester tester) async {
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new ListWheelScrollView(
+          itemExtent: 50.0,
+          children: <Widget>[
+            const SizedBox(
+              height: 200.0,
+              width: 200.0,
+              child: const Center(
+                child: const Text('blah'),
+              ),
+            ),
+          ],
+        ),
+      )
+    );
+    expect(tester.getSize(find.byType(SizedBox)), const Size(200.0, 50.0));
+    expect(find.text('blah'), findsOneWidget);
```

Comment:
btw, you get bonus points for using obscure references (e.g. to your favourite musical or movie) instead of the boring "blah". :-)

Reply:
Ha. I'll keep it in mind.